ydb/core/blobstorage/dsproxy/dsproxy_blackboard.cpp

```diff
@@ -542,4 +542,35 @@ void TBlackboard::InvalidatePartStates(ui32 orderNumber) {
     }
 }
 
+void TBlackboard::MarkSlowDisks(TBlobState& state, bool isPut, const TAccelerationParams& accelerationParams) {
+    // by default all disks are considered fast
+    for (TBlobState::TDisk& disk : state.Disks) {
+        disk.IsSlow = false;
+    }
+
+    ui32 maxNumSlow = accelerationParams.MaxNumOfSlowDisks;
+    if (Info->GetTotalVDisksNum() <= maxNumSlow) {
+        // all disks cannot be slow
+        return;
+    }
+    
+    TDiskDelayPredictions worstDisks;
+    state.GetWorstPredictedDelaysNs(*Info, *GroupQueues,
+            (isPut ? HandleClassToQueueId(PutHandleClass) : HandleClassToQueueId(GetHandleClass)),
+            &worstDisks, accelerationParams.PredictedDelayMultiplier);
+
+    ui64 slowThreshold = worstDisks[maxNumSlow].PredictedNs * accelerationParams.SlowDiskThreshold;
```

Comment:
In case of 2 slow disks this would be the third slow from the top, correct? It seems a bit strange that we calculate threshold instead of just marking N top slowest disks.

Reply:
We only want to mark N slowest disks as "slow", if they are at least twice as slow as worst "normal" disk, that's why we need a threshold